### DISCUSSION TITLE
[Question]: Agents not working

### DISCUSSION THREAD

**Discussion by MikeNatC** [danny-avila/LibreChat#4902] [Troubleshooting] [answered]
### What is your question?

The Agents function does not work with Open AI and Gemini models although it works with Anthrophic's models.

### More Details

I am running the latest release of LibreChat v0.7.6-rc1. 

I get the following error when chatting with an Agent built with Open AI models: "No key found. Please provide a key and try again."

When chatting with an agent built with the Google gemini-pro model, I get this error: "Something went wrong. Here's the specific error message we encountered: An error occurred while processing your request. Please contact the Admin."

However, I am able to chat with an Agent built with Anthrophic's models. 

I am not sure what is causing this issue. I have an Open AI key which input in the Open AI custom end point unless there is somewhere else to input an Open AI key for Agents. I don't have any idea what is causing the issue with Agents running Google's model. Would appreciate any help I can get on this issue!

### What is the main subject of your question?

Other

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Google endpoints are not supported yet but will be with v0.7.6 proper. Is the API key provided from the UI? I think I know the issue there.

**Reply by MikeNatC** [reply]
Yes, the API key for Open AI is provided through the user interface in the endpoint drop down menu. The same as for the Anthrophic model.

**Reply by danny-avila** [reply, marked as answer]
Ok, I merged a fix here, should work after, tested on my end: https://github.com/danny-avila/LibreChat/pull/4903

**Reply by MikeNatC** [reply]
Thank you so much for the quick fix!

**Reply by danny-avila** [reply]
Google agents support: https://github.com/danny-avila/LibreChat/pull/5126

**Comment by phalcomb** [comment]
Will agents be supported for ollama as well?

**Reply by danny-avila** [reply]
Yes, it is already supported

**Reply by phalcomb** [reply]
Thanks, I must have turned it off somehow, I've scoured the docs and can't seem to find where to turn the feature on. I have only one EndPoint setup in my librechat.yaml file and that's for ollama. BTW, I'm using the latest Docker build, also thanks for all the work with this, I'm having a lot of fun with it.

**Reply by phalcomb** [reply]
Disregard, I figured it out, I limited my endpoints to Ollama in my .env file. I just added _agents_ to the list and that resolved the issue. Thanks again, really nice program.

**Reply by LeonardSEO** [reply]
@danny-avila 
I have this problem with Azure OpenAI. Are agents supported for Azure OpenAI?  

ERROR:  
`Something went wrong. Here's the specific error message we encountered: No model spec selected
`
